kiwi/xml_state.py

```diff
@@ -623,6 +623,17 @@ def get_build_type_spare_part_size(self):
         if spare_part_size:
             return self._to_mega_byte(spare_part_size)
 
+    def get_build_type_mediacheck(self):
+        """
+        Get the mediacheck attribute from the build type. If
+        no mediacheck is present returns False.
+
+        :rtype: bool
+        """
+        if self.build_type.get_mediacheck() is True:
+            return True
+        return False
```

Comment:
Just an idea and I leave it up to you if you want to change it:

The mediacheck attribute is specified as xsd:boolean, thus you can rely on the generated api to provide a True, False or None(if not specified) value. As the arch constraint is covered by the runtime test it would be possible to drop the method and directly read the information from xml_state.build_type.get_mediacheck(). As long as None(not specified) and False can be treated the same way which is the case here this could be done